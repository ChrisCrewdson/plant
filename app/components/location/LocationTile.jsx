// Used to show a summary tile/card of a Location

const React = require('react');
const { Link } = require('react-router-dom');
const PropTypes = require('prop-types');
const utils = require('../../libs/utils');
const { actionFunc } = require('../../actions');

const { makeSlug } = utils;

/**
 *
 * @param {string} _id
 * @param {import('redux').Dispatch} dispatch
 */
function onLinkClick(_id, dispatch) {
  dispatch(actionFunc.changeActiveLocationId({ _id }));
}

function locationTile(props) {
  const {
    _id,
    dispatch,
    numPlants,
    title,
  } = props;

  const link = `/location/${makeSlug(title)}/${_id}`;
  const locationTitle = `${title} (${numPlants})`;

  const style = {
    display: 'flex',
    alignItems: 'center',
  };

  return (
    <div style={style}>
      <Link
        style={{ margin: '20px' }}
        to={link}
        onClick={() => { onLinkClick(_id, dispatch); }}
      >
        <span>
          {locationTitle}
        </span>
      </Link>
    </div>
  );
}

locationTile.propTypes = {
  _id: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired,
  numPlants: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
};

module.exports = locationTile;
