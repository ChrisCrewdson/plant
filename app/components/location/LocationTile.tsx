// Used to show a summary tile/card of a Location

import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Dispatch } from 'redux';

import utils from '../../libs/utils';
import { actionFunc } from '../../actions';
import { PlantAction, ChangeActiveLocationIdPayload } from '../../../lib/types/redux-payloads';

const { makeSlug } = utils;

interface LocationTileProps {
  _id: string;
  dispatch: Dispatch<PlantAction<ChangeActiveLocationIdPayload>>;
  numPlants: number;
  title: string;
}

function onLinkClick(_id: string, dispatch: Dispatch<PlantAction<ChangeActiveLocationIdPayload>>) {
  const payload: ChangeActiveLocationIdPayload = { _id };
  dispatch(actionFunc.changeActiveLocationId(payload));
}

export default function locationTile(props: LocationTileProps) {
  const {
    _id,
    dispatch,
    numPlants,
    title,
  } = props;

  const link = `/location/${makeSlug(title)}/${_id}`;
  const locationTitle = `${title} (${numPlants})`;

  const style = {
    display: 'flex',
    alignItems: 'center',
  };

  return (
    <div style={style}>
      <Link
        style={{ margin: '20px' }}
        to={link}
        onClick={() => { onLinkClick(_id, dispatch); }}
      >
        <span>
          {locationTitle}
        </span>
      </Link>
    </div>
  );
}

locationTile.propTypes = {
  _id: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired,
  numPlants: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
};
