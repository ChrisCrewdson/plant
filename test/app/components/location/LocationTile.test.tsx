import { Dispatch } from 'redux';

import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import { MemoryRouter } from 'react-router-dom';
import LocationTile from '../../../../app/components/location/LocationTile';

const muiTheme = getMuiTheme(lightBaseTheme);

const mockDispatch = (() => {}) as Dispatch;

describe('LocationTile', () => {
  test('LocationTile should be rendered', () => {
    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <MemoryRouter>
          <LocationTile
            _id="mongo-id"
            numPlants={5}
            dispatch={mockDispatch}
            title="I am a title"
          />
        </MemoryRouter>
      </MuiThemeProvider>);
    const badgeTotals = component.toJSON();
    expect(badgeTotals).toMatchSnapshot();
  });
});
