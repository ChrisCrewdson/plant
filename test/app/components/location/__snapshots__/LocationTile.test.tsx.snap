// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationTile LocationTile should be rendered 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
>
  <a
    href="/location/i-am-a-title/mongo-id"
    onClick={[Function]}
    style={
      Object {
        "margin": "20px",
      }
    }
  >
    <span>
      I am a title (5)
    </span>
  </a>
</div>
`;
